authority/provisioner/k8sSA.go

```diff
@@ -0,0 +1,263 @@
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+
+	"github.com/pkg/errors"
+	"github.com/smallstep/cli/crypto/pemutil"
+	"github.com/smallstep/cli/jose"
+	"golang.org/x/crypto/ed25519"
+)
+
+// NOTE: There can be at most one kubernetes service account provisioner configured
+// per instance of step-ca. This is due to a lack of distinguishing information
+// contained in kubernetes service account tokens.
+
+const (
+	// K8sSAName is the default name used for kubernetes service account provisioners.
+	K8sSAName = "k8sSA-default"
+	// K8sSAID is the default ID for kubernetes service account provisioners.
+	K8sSAID     = "k8ssa/" + K8sSAName
+	k8sSAIssuer = "kubernetes/serviceaccount"
+)
+
+// This number must <= 1. We'll verify this in Init() below.
+var numK8sSAProvisioners = 0
+
+// jwtPayload extends jwt.Claims with step attributes.
+type k8sSAPayload struct {
+	jose.Claims
+	Namespace          string `json:"kubernetes.io/serviceaccount/namespace,omitempty"`
+	SecretName         string `json:"kubernetes.io/serviceaccount/secret.name,omitempty"`
+	ServiceAccountName string `json:"kubernetes.io/serviceaccount/service-account.name,omitempty"`
+	ServiceAccountUID  string `json:"kubernetes.io/serviceaccount/service-account.uid,omitempty"`
+}
+
+// K8sSA represents a Kubernetes ServiceAccount provisioner; an
+// entity trusted to make signature requests.
+type K8sSA struct {
+	Type      string  `json:"type"`
+	Name      string  `json:"name"`
+	Claims    *Claims `json:"claims,omitempty"`
+	PubKeys   []byte  `json:"pubKeys,omitempty"`
+	claimer   *Claimer
+	audiences Audiences
+	//kauthn    kauthn.AuthenticationV1Interface
+	pubKeys []interface{}
+}
+
+// GetID returns the provisioner unique identifier. The name and credential id
+// should uniquely identify any K8sSA provisioner.
+func (p *K8sSA) GetID() string {
+	return K8sSAID
+}
+
+// GetTokenID returns an unimplemented error and does not use the input ott.
+func (p *K8sSA) GetTokenID(ott string) (string, error) {
+	return "", errors.New("not implemented")
+}
+
+// GetName returns the name of the provisioner.
+func (p *K8sSA) GetName() string {
+	return p.Name
+}
+
+// GetType returns the type of provisioner.
+func (p *K8sSA) GetType() Type {
+	return TypeK8sSA
+}
+
+// GetEncryptedKey returns false, because the kubernetes provisioner does not
+// have access to the private key.
+func (p *K8sSA) GetEncryptedKey() (string, string, bool) {
+	return "", "", false
+}
+
+// Init initializes and validates the fields of a K8sSA type.
+func (p *K8sSA) Init(config Config) (err error) {
+	switch {
+	case p.Type == "":
+		return errors.New("provisioner type cannot be empty")
+	case p.Name == "":
+		return errors.New("provisioner name cannot be empty")
+	// Verify that any kubernetes service account provisioner has the standard,
+	// default name.
+	case p.Name != K8sSAName:
+		return errors.Errorf("k8sSA provisioner name must be %s, but got %s", K8sSAName, p.Name)
```

Comment:
I would let the user let to use any name, and just use the name on `GetID()` used in `Collection.Store(p Interface)`